### PR TITLE
cherry-pick  for Hotplug RHEL8.x issue 

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
Cherry-pick from https://github.com/Xilinx/XRT/pull/5688

### CODE REVIEW

**Comment by xilinxgitops** [comment]
retest this please

File: src/runtime_src/core/pcie/linux/scan.cpp [hunk 1]
```diff
@@ -1149,6 +1149,37 @@ get_uuids(std::shared_ptr<char>& dtbbuf, std::vector<std::string>& uuids)
...
+ */
+int
+get_runtime_active_kids(std::string &pci_bridge_path)
+{
+  int curr_act_dev = 0;
+  std::vector<bfs::path> vec{bfs::directory_iterator(pci_bridge_path), bfs::directory_iterator()};
+
+  // Check number of Xilinx devices under this bridge. 
+  for (auto& path : vec) {
+    if (bfs::is_directory(path)) {
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.cpp:1165]: Use early exit from the loop by inverting this check

Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/linux/scan.cpp]: Modified. Thanks.

**Comment by dayeh-xilinx** [comment]
retest this please. Debugging the hardware test script which fails u30

**Comment by dayeh-xilinx** [comment]
retest this please. Debugging the hardware test script which fails u30

**Comment by dayeh-xilinx** [comment]
retest this please. Debugging the hardware test script which fails u30

**Comment by xilinxgitops** [comment]
retest this please

**Comment by xilinxgitops** [comment]
retest this please

**Pull request merged**